src/inferenceql/query/scalar.cljc

```diff
@@ -77,8 +78,7 @@
     [:conditioned-by-expr model _conditioned _by event] `(~'iql/condition ~(plan model) ~(plan event))
     [:constrained-by-expr model _constrained _by event] `(~'iql/constrain ~(plan model) ~(plan event))
 
-    [:value [:null _]] nil
-    [:value [_ child]] (edn/read-string child)
+    [:value child] (literal/read child)
 
```

Comment:
This might represent unnecessary work, but should be safe. While `literal/read` can match a whole lot more than literals, only the handful of possibilities under `value` should be of any concern.